transport/transporters/kafka/client_config.yaml.go

```diff
@@ -0,0 +1,126 @@
+		"running on a local machine (not in Docker) needs to communicate with a Kafka cluster in Kubernetes in Docker.")
+
+	clusterCA        = flag.String("kafka_cluster_ca", "/ssl/cluster/ca.crt", "Path to certificate for the cluster CA.")
+	clientPrivateKey = flag.String("kafka_client_key_file", "/ssl/client/user.key", "Path to client's private key used for authentication.")
+	clientPublicKey  = flag.String("kafka_client_cert_file", "/ssl/client/user.crt", "Path to client's certificate containing the client's public key.")
+)
+
+func producerConfig() (*sarama.Config, error) {
+	// TODO: Set up a custom logger for Sarama debug logging.
+	//sarama.Logger = clientLogger(true)
+
+	// We don't have a registry set up, so might as well disable these.
+	metrics.UseNilMetrics = true
+
+	config := sarama.NewConfig()
+	config.Version = sarama.V3_0_0_0
+
+	// Default client value. This should be kept lower, since we don't have control over
+	// Sarama's internal buffering. Having too high of a value may cause shutdown to be
+	// blocked for too long.
+	config.ChannelBufferSize = 256
+
+	// Default value is 30 seconds, which is too high. We haven't actively tried to optimize these values.
+	config.Net.DialTimeout = 10 * time.Second
+	config.Net.ReadTimeout = 10 * time.Second
+	config.Net.WriteTimeout = 10 * time.Second
+
+	// This produces reasonably efficient batching, but could be tuned more.
+	config.Producer.Flush.Bytes = 262144
+
+	// Should be less than `closeDelay` to make a best-effort attempt at flushing
+	// all messages when shutting down.
+	config.Producer.Flush.Frequency = 2500 * time.Millisecond
+
+	// This is required to track metrics for successful writes.
+	config.Producer.Return.Successes = true
+
+	// This is required to track any errors.
+	config.Producer.Return.Errors = true
+
+	// We previously used ztsd, which produced a better compression ratio, but the version used by
+	// Sarama 1.29 and 1.30 would use way too much memory over time: https://github.com/Shopify/sarama/issues/1831.
+	config.Producer.Compression = sarama.CompressionSnappy
+
+	// Default client value.
+	config.Producer.MaxMessageBytes = 1000000
+
+	// Default value is 100ms. During cluster restarts, we're at risk of hitting the max retry limit
+	// for a partition if we retry too quickly. This value was increased when debugging buffer flushing
+	// issues, but hasn't been actively tuned / optimized.
+	config.Producer.Retry.Backoff = 500 * time.Millisecond
+
+	// Reduce memory usage by only holding onto the metadata needed for GoFlask's topic(s).
+	config.Metadata.Full = false
+
+	// By default, Kafka closes idle connections after 10 minutes.
+	// For brokers that don't have any GoFlask partitions, the connection is idle other
+	// than metadata refreshing, so at times the client will try to use a connection
+	// that the broker has closed.
+	config.Metadata.RefreshFrequency = 5 * time.Minute
```

Comment:
How does this relate to the retry in our application code? Should we configure saram to not retry?

Reply:
I think we should configure sarama to handle retrying instead of the application, unless you have strong opinions against it.